pkg/operator/operator.go

```diff
@@ -811,6 +812,38 @@ func (op *Reporting) newPrometheusConn(promConfig promapi.Config) (prom.API, err
 	return prom.NewAPI(client), nil
 }
 
+func (op *Reporting) handleExpiredReport(report *metering.Report, now time.Time) error {
+	// check if the Report is being deleted already
+	if report.DeletionTimestamp != nil {
+		return fmt.Errorf("report was already marked for deletion")
+	}
+	// check if the Report is past its retention time
+	if reportExpired := isReportExpired(op.logger, report, now); !reportExpired {
+		op.logger.Debugf("the %s Report in the %s namespace has not reached the expiration date", report.Name, report.Namespace)
+		return nil
+	}
+	// check if the Report is used by any other Report or ReportQuery, if not delete it
+	if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+		newDeletionTimestamp := metav1.Now()
+		report.SetDeletionTimestamp(&newDeletionTimestamp)
```

Comment:
Hmm, I'm still not 100% sure we should be setting these values. Is it not sufficient to just to make the API call to delete this resource?

Reply:
https://kubernetes.io/docs/reference/using-api/api-concepts/ it's an async process setting it begins the process of gettting rid of the resource with all the finalizers being allowed to act once you set it. In this case I think it's okay to set and then have the client delete the resource because we actually check it ourselves at the top of this function as a way to prevent wasting time reattempting the delete.